Lib/test/test_exceptions.py

```diff
@@ -318,11 +319,17 @@ def baz():
         check('(yield i) = 2', 1, 2)
         check('def f(*):\n  pass', 1, 7)
 
+    @unittest.skipIf(ctypes.sizeof(ctypes.c_ssize_t) > ctypes.sizeof(ctypes.c_int),
+                     "Py_ssize_t is bigger than INT_MAX, so this is unreachable")
     @support.requires_resource('cpu')
     @support.bigmemtest(support._2G, memuse=1.5)
     def testMemoryErrorBigSource(self):
-        with self.assertRaises(OverflowError):
-            exec(f"if True:\n {' ' * 2**31}print('hello world')")
+        # the padding length needs to be more than INT_MAX, but we can't
+        # multiple a sequence by a number that doesn't fit Py_ssize_t,
+        # otherwise we will get an OverflowError (see PySequence_Repeat)
+        padding = '       ' * ((ctypes.sizeof(ctypes.c_int) + 1) / 8)
```

Comment:
I don't think this tests the correct thing. The buffer is way too small to overflow, probably a typo?

Reply:
Yep, I forgot it returns the number of bytes :sweat_smile: